Problems/prob031/models/RackConfiguration.essence

```diff
@@ -31,7 +31,7 @@ given max_power    : function (total) Model --> int(1..),
 $ model: the model of each rack
 $ plugged: the set of cards plugged in to a particular rack
 find model   : function Rack --> Model,
-     plugged : function Rack --> set of Card
+     plugged : function Rack --> set (maxSize max(range(max_connects))) of Card
```

Comment:
The constraint:

```essence
forAll r in defined(model) .
         |plugged(r)| <= max_connects(model(r))
```

constrains the size of the set returned by the `plugged` function to a value in `range(max_connects)`. Because `max_connects` is a `given` function, its range will be fully known. The `max` of the range is taken as it is called with different parameters in a `forAll`.